### PR TITLE
[DEMO] Major diktat-demo improvements

### PR DESCRIPTION
**Pull request by sanyavertolet**
### What's done:
 * Fixed an issue with crashes of diktat fix - set env's `JAVA_TOOL_OPTIONS` and `_JAVA_OPTIONS` to empty string
 * Added `Demo` dropdown to `top bar` at frontend
 * Improved diktat temp configuration - now there always is default `diktat-analysis.yaml` and `src` is in path to working dir
 * Added `Constants.kt` file with demo's constants

### CODE REVIEW

File: save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/demo/diktat/DemoAdditionalParams.kt [hunk 1]
```diff
@@ -14,6 +14,109 @@ import kotlinx.serialization.Serializable
...
     val mode: DiktatDemoMode = DiktatDemoMode.WARN,
     val tool: DiktatDemoTool = DiktatDemoTool.DIKTAT,
-    val config: String? = null,
-    val language: Languages = Languages.KOTLIN
-) : DemoAdditionalParams
+    val config: List<String> = defaultDiktatConfig,
+    val language: Languages = Languages.KOTLIN,
+) : DemoAdditionalParams {
+    companion object {
+        val defaultDiktatConfig = """
```
Comment (nulls) [inline save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/demo/diktat/DemoAdditionalParams.kt:21]: maybe extract to a file in /resources?

Comment (0x6675636b796f75676974687562) [inline save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/demo/diktat/DemoAdditionalParams.kt:21]:
```suggestion
        @Language("YAML")
        val defaultDiktatConfig = """
```

Bacisally, both `YaMl` and `yMl` (in any case) work.

**Review by 0x6675636b796f75676974687562 — APPROVED** [review]

File: save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/demo/diktat/DemoAdditionalParams.kt (hunk 1 above)
Comment (sanyavertolet) [reply save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/demo/diktat/DemoAdditionalParams.kt:21]: I think of adding it to storage, but in this case I will need to get it from storage if and only if there is no config provided. Will be done later in next PRs

Comment (sanyavertolet) [reply save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/demo/diktat/DemoAdditionalParams.kt:21]: sure, thanks

Comment (sanyavertolet) [reply save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/demo/diktat/DemoAdditionalParams.kt:22]: Seems that commonMain does not have `@Language` annotation so will remove as was

Comment (petertrr) [reply save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/demo/diktat/DemoAdditionalParams.kt:21]: https://www.jetbrains.com/help/idea/using-language-injections.html#use-language-injection-comments

**Pull request merged**